instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/Network.kt

```diff
@@ -2,82 +2,84 @@ package com.mapbox.navigation.instrumentation_tests.utils
         uiAutomation.executeShellCommand("svc data enable")
     }
-    Log.d(LOG_TAG, "Waiting for network to become reachable")
-    reachability.waitForNetworkStatus { it != NetworkStatus.NOT_REACHABLE }
+    assumeNetworkIsReachable()
 }
 
-private suspend fun assumeNetworkIsNotReachable(reachability: ReachabilityInterface) {
-    val networkIsReachable = withTimeoutOrNull(3000) {
-        reachability.waitForNetworkStatus { it == NetworkStatus.NOT_REACHABLE }
-        false
-    } ?: true
-    assumeFalse(
+private suspend fun assumeNetworkIsNotReachable() {
+    val result = waitForHostReachability(expectedReachability = false)
+    assumeTrue(
         "network should not be reachable if it's turned off on device",
-        networkIsReachable
+        result == WaitResult.SUCCESS
     )
 }
 
-private suspend fun ReachabilityInterface.waitForNetworkStatus(
-    condition: (NetworkStatus) -> Boolean
-) {
-    val currentStatus = currentNetworkStatus()
-    if (condition(currentNetworkStatus())) {
-        Log.d(LOG_TAG, "Network status $currentStatus is ok")
-        return
-    }
-    suspendCancellableCoroutine<Unit> { continuation ->
-        val id = this.addListener { currentStatus ->
-            val satisfiesCondition = condition(currentStatus)
-            val messageForStatus = if (satisfiesCondition) {
-                "Ok."
-            } else {
-                "Keep on waiting for updates."
-            }
-            Log.d("Network", "Current network status $currentStatus. $messageForStatus")
-            if (satisfiesCondition) {
-                if (continuation.isActive) {
-                    continuation.resume(Unit)
+private suspend fun assumeNetworkIsReachable() {
+    val result = waitForHostReachability(expectedReachability = true)
+    assumeTrue(
+        "host should be reachable when network is turned on",
+        result == WaitResult.SUCCESS
+    )
+}
+
+private enum class WaitResult { SUCCESS, TIMEOUT }
+
+private suspend fun waitForHostReachability(expectedReachability: Boolean): WaitResult =
+    withTimeoutOrNull(10_000) {
+        withContext(Dispatchers.IO) {
+            while (true) {
+                val actualReachability = checkReachability()
+                Log.d(LOG_TAG, "is host reachable: $actualReachability, waiting for $expectedReachability")
+                if (actualReachability == expectedReachability) {
+                    break
+                } else {
+                    delay(500)
                 }
             }
+            WaitResult.SUCCESS
         }
-        continuation.invokeOnCancellation {
-            this.removeListener(id)
-        }
-    }
+    } ?: WaitResult.TIMEOUT
+
+private fun checkReachability(): Boolean {
+    val command = "ping -W 1 -c 1 8.8.8.8"
+    val result = InstrumentationRegistry.getInstrumentation()
+        .uiAutomation
+        .executeShellCommandBlocking(command)
+        .let { String(it) }
```

Comment:
I'm not sure about stability of this approach, because there are could be devices without `ping` installed or with a different one. It works stable on 3 devices we use in test lab, but I would rather prefer java implementation of ping mechanism to be sure that it works on any device